test/functional/images_controller_test.rb

```diff
@@ -49,7 +49,7 @@ class ImagesControllerTest < ActionController::TestCase
     # This value is tested by the deprecation warning in application_controller
     # so we need to set it or the test will crash
     request.env['REQUEST_URI']="compute_resources/#{@image.compute_resource_id}/images"
-    get :index, { :format => :json, :compute_resource_id => @image.compute_resource_id }, set_session_user
+    get :index, { :format => :json, :compute_resource_id => @image.compute_resource_id, :operatingsystem_id => @image.operatingsystem_id, :architecture_id => @image.architecture_id }, set_session_user
```

Comment:
What about adding another image for different OS into fixtues, just to check that the filter will only match the images the for this operatingsystem: currently it would be green even when the os filter was not there at all.

```
three:
  operatingsystem: redhat
  architecture: x86_64
  compute_resource: one
  uuid: RedHatImage
  username: MyString
  name: redhat-1
```